sonar-application/src/main/java/org/sonar/application/Cluster.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * SonarQube
+ * Copyright (C) 2009-2017 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+package org.sonar.application;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.hazelcast.config.Config;
+import com.hazelcast.config.JoinConfig;
+import com.hazelcast.config.NetworkConfig;
+import com.hazelcast.core.Hazelcast;
+import com.hazelcast.core.HazelcastInstance;
+import java.util.Arrays;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.sonar.process.Props;
+
+/**
+ * Manager for the cluster communication between Main Processes
+ */
+public final class Cluster {
+  private static final Logger LOG = LoggerFactory.getLogger(Cluster.class);
+  /**
+   * The Hazelcast instance.
+   */
+  @VisibleForTesting
+  public HazelcastInstance hazelcastInstance;
+
+  private boolean active = false;
+
+  /**
+   * Instantiates a new Cluster.
+   *
+   * @param props the props
+   */
+  protected Cluster(Props props) {
+    ClusterProperties.completeDefaults(props);
+
+    if (props.valueAsBoolean(ClusterProperties.ENABLED)) {
+      initializeHazelcast(props);
+    }
+  }
+
+  private void initializeHazelcast(Props props) {
+    Config hzConfig = new Config();
+    // Configure the network instance
+    NetworkConfig netConfig = hzConfig.getNetworkConfig();
+    netConfig.setPort(props.valueAsInt(ClusterProperties.PORT))
+      .setPortAutoIncrement(props.valueAsBoolean(ClusterProperties.PORT_AUTOINCREMENT));
+
+    if (props.value(ClusterProperties.INTERFACES) != "") {
+      netConfig.getInterfaces()
+        .setEnabled(true)
+        .setInterfaces(
+          Arrays.asList(props.value(ClusterProperties.INTERFACES).split(","))
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') NullPointerException might be thrown as 'value' is nullable here [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2259)